### PR TITLE
Fix ID and IDReadonly.

### PR DESCRIPTION
**Pull request by martinpovolny**
The code generators do not expect models to be non-objects.

Fixes https://github.com/ManageIQ/topological_inventory-api/issues/123

might also fix https://github.com/ManageIQ/topological_inventory-api-client-ruby/pull/3

Ping @agrare, @bdunne, @eclarizio, @Hyperkid123 

Generating the javascript client from fixed swagger file w/o any issues and hand-editing:

```
lucifer - [~/Projects/sources/topological_inventory-api-jsclient] (master)$ npm run build

> @manageiq/topological_inventory@1.0.6 build /home/martin/Projects/sources/topological_inventory-api-jsclient
> webpack --config ./conf/webpack.config.js

Hash: 6a51c643a4d9b97b7a05
Version: webpack 4.28.1
Time: 1618ms
Built at: 2019-02-28 14:09:54
  Asset     Size  Chunks             Chunk Names
main.js  608 KiB    main  [emitted]  main
Entrypoint main = main.js
[./src/ApiClient.js] 20.1 KiB {main} [built]
[./src/api/DefaultApi.js] 161 KiB {main} [built]
[./src/index.js] 10.4 KiB {main} [built]
[./src/model/Authentication.js] 4.63 KiB {main} [built]
[./src/model/AuthenticationsCollection.js] 3.46 KiB {main} [built]
[./src/model/AvailabilitiesCollection.js] 3.43 KiB {main} [built]
[./src/model/Availability.js] 4.66 KiB {main} [built]
[./src/model/CollectionLinks.js] 3.24 KiB {main} [built]
[./src/model/CollectionMetadata.js] 2.68 KiB {main} [built]
[./src/model/Container.js] 4.85 KiB {main} [built]
[./src/model/ContainerGroup.js] 6.01 KiB {main} [built]
[./src/model/ContainerGroupsCollection.js] 3.46 KiB {main} [built]
[./src/model/ContainerImage.js] 5.44 KiB {main} [built]
[./src/model/ContainerImagesCollection.js] 3.46 KiB {main} [built]
[./src/model/ContainerNode.js] 6.11 KiB {main} [built]
    + 55 hidden modules
```

I believe the issues with the Ruby client will also be fixed by this change.

### CODE REVIEW

**Comment by Ladas** [comment]
@martinpovolny could you update the generator like @bdunne has done here https://github.com/ManageIQ/topological_inventory-api/pull/140? If we are removing also the `ID`. What @bdunne is changing there is the only code that generates the `ID` related stuff, so we can change that to naything.

Then generate the swagger.yaml with `rake swagger:generate`

**Comment by eclarizio** [comment]
As @bdunne mentioned on standup I had a similar problem with https://github.com/ManageIQ/topological_inventory-api/pull/131, but that was because of the way IDReadOnly was nested and the generator failing to figure out how to process that. We determined that IDs should all just be read only anyway, so my question is does the javascript generator still have problems figuring it out if we just use the `"$ref": "#/definitions/ID"` with the readOnly property set? It would really suck if we had to duplicate the regex and other properties everywhere we need an ID.

File: public/doc/swagger-2-v0.1.0.yaml [hunk 1]
```diff
@@ -1406,14 +1406,17 @@ definitions:
         example: openshift_default
         type: string
       id:
-        "$ref": "#/definitions/IDReadOnly"
+        type: string
+        pattern: "/^\\d+$/"
```
Comment (bdunne) [inline public/doc/swagger-2-v0.1.0.yaml]: I'd prefer not to duplicate this everywhere, is the issue that `IDReadOnly` depended on `ID`?  (that sounds like a bug in the code generator)
Hopefully this problem is resolved by #140

**Comment by martinpovolny** [comment]
Just retested with the swagger file from #140 and seems the problem is really fixed :+1: 

So I can close this one.

I will update https://github.com/ManageIQ/topological_inventory-api-jsclient/pull/5 (well maybe the files will get generated the same, will check).

Anyway seems we can close https://github.com/ManageIQ/topological_inventory-api/issues/123 now.

**Comment by Hyperkid123** [comment]
I did test it right now and with #140 the code gets generated and the client builds correctly.

**Comment by martinpovolny** [comment]
Also the generated javascript source is the same except for the "ID of the resource" comment. So we should be good!

**Comment by agrare** [comment]
Okay awesome I'll regenerate the topological_inventory-api-client-ruby to fix https://github.com/ManageIQ/topological_inventory-api-client-ruby/pull/3

**Pull request closed without merging**